clients/src/test/java/org/apache/kafka/common/record/DefaultRecordBatchTest.java

```diff
@@ -63,15 +61,9 @@
 import static org.mockito.Mockito.when;
 
 public class DefaultRecordBatchTest {
-    private static final Random RANDOM;
-
-    static {
-        try {
-            RANDOM = SecureRandom.getInstanceStrong();
-        } catch (NoSuchAlgorithmException e) {
-            throw new RuntimeException(e);
-        }
-    }
+    // We avoid SecureRandom.getInstanceStrong() here because it reads from /dev/random and blocks on Linux. Since these
+    // tests don't require cryptographically strong random data, we avoid a CSPRNG (SecureRandom) altogether.
+    private static final Random RANDOM = new Random();
```

Comment:
can you please provide a constant seed here so that the tests are reproducible and deterministic?

Reply:
Addressed in [afc7129](https://github.com/apache/kafka/pull/14625/commits/afc71297c805eb149a3dee450cd4e204c4fb57d9)